// @azure/ai-language-text/dist-esm/src/models.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.
import { KnownErrorCode, KnownInnerErrorCode, } from "./generated";
/**
 * Type of actions supported by the {@link TextAnalysisClient.analyze} method.
 */
export const AnalyzeActionNames = {
    EntityLinking: "EntityLinking",
    EntityRecognition: "EntityRecognition",
    KeyPhraseExtraction: "KeyPhraseExtraction",
    PiiEntityRecognition: "PiiEntityRecognition",
    LanguageDetection: "LanguageDetection",
    SentimentAnalysis: "SentimentAnalysis",
};
/**
 * Type of actions supported by the {@link TextAnalysisClient.beginAnalyzeBatch} method.
 */
export const AnalyzeBatchActionNames = {
    SentimentAnalysis: "SentimentAnalysis",
    EntityRecognition: "EntityRecognition",
    PiiEntityRecognition: "PiiEntityRecognition",
    KeyPhraseExtraction: "KeyPhraseExtraction",
    EntityLinking: "EntityLinking",
    Healthcare: "Healthcare",
    ExtractiveSummarization: "ExtractiveSummarization",
    AbstractiveSummarization: "AbstractiveSummarization",
    CustomEntityRecognition: "CustomEntityRecognition",
    CustomSingleLabelClassification: "CustomSingleLabelClassification",
    CustomMultiLabelClassification: "CustomMultiLabelClassification",
};
/**
 * Enum of possible error codes of a {@link TextAnalysisError}.
 */
export const KnownTextAnalysisErrorCode = Object.assign(Object.assign({}, KnownErrorCode), KnownInnerErrorCode);
