// @azure/ai-language-text/dist-esm/src/generated/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link AnalyzeTextTaskKind} that the service accepts. */
export var KnownAnalyzeTextTaskKind;
(function (KnownAnalyzeTextTaskKind) {
    /** SentimentAnalysis */
    KnownAnalyzeTextTaskKind["SentimentAnalysis"] = "SentimentAnalysis";
    /** EntityRecognition */
    KnownAnalyzeTextTaskKind["EntityRecognition"] = "EntityRecognition";
    /** PiiEntityRecognition */
    KnownAnalyzeTextTaskKind["PiiEntityRecognition"] = "PiiEntityRecognition";
    /** KeyPhraseExtraction */
    KnownAnalyzeTextTaskKind["KeyPhraseExtraction"] = "KeyPhraseExtraction";
    /** LanguageDetection */
    KnownAnalyzeTextTaskKind["LanguageDetection"] = "LanguageDetection";
    /** EntityLinking */
    KnownAnalyzeTextTaskKind["EntityLinking"] = "EntityLinking";
})(KnownAnalyzeTextTaskKind || (KnownAnalyzeTextTaskKind = {}));
/** Known values of {@link AnalyzeTextTaskResultsKind} that the service accepts. */
export var KnownAnalyzeTextTaskResultsKind;
(function (KnownAnalyzeTextTaskResultsKind) {
    /** SentimentAnalysisResults */
    KnownAnalyzeTextTaskResultsKind["SentimentAnalysisResults"] = "SentimentAnalysisResults";
    /** EntityRecognitionResults */
    KnownAnalyzeTextTaskResultsKind["EntityRecognitionResults"] = "EntityRecognitionResults";
    /** PiiEntityRecognitionResults */
    KnownAnalyzeTextTaskResultsKind["PiiEntityRecognitionResults"] = "PiiEntityRecognitionResults";
    /** KeyPhraseExtractionResults */
    KnownAnalyzeTextTaskResultsKind["KeyPhraseExtractionResults"] = "KeyPhraseExtractionResults";
    /** LanguageDetectionResults */
    KnownAnalyzeTextTaskResultsKind["LanguageDetectionResults"] = "LanguageDetectionResults";
    /** EntityLinkingResults */
    KnownAnalyzeTextTaskResultsKind["EntityLinkingResults"] = "EntityLinkingResults";
})(KnownAnalyzeTextTaskResultsKind || (KnownAnalyzeTextTaskResultsKind = {}));
/** Known values of {@link ErrorCode} that the service accepts. */
export var KnownErrorCode;
(function (KnownErrorCode) {
    /** InvalidRequest */
    KnownErrorCode["InvalidRequest"] = "InvalidRequest";
    /** InvalidArgument */
    KnownErrorCode["InvalidArgument"] = "InvalidArgument";
    /** Unauthorized */
    KnownErrorCode["Unauthorized"] = "Unauthorized";
    /** Forbidden */
    KnownErrorCode["Forbidden"] = "Forbidden";
    /** NotFound */
    KnownErrorCode["NotFound"] = "NotFound";
    /** ProjectNotFound */
    KnownErrorCode["ProjectNotFound"] = "ProjectNotFound";
    /** OperationNotFound */
    KnownErrorCode["OperationNotFound"] = "OperationNotFound";
    /** AzureCognitiveSearchNotFound */
    KnownErrorCode["AzureCognitiveSearchNotFound"] = "AzureCognitiveSearchNotFound";
    /** AzureCognitiveSearchIndexNotFound */
    KnownErrorCode["AzureCognitiveSearchIndexNotFound"] = "AzureCognitiveSearchIndexNotFound";
    /** TooManyRequests */
    KnownErrorCode["TooManyRequests"] = "TooManyRequests";
    /** AzureCognitiveSearchThrottling */
    KnownErrorCode["AzureCognitiveSearchThrottling"] = "AzureCognitiveSearchThrottling";
    /** AzureCognitiveSearchIndexLimitReached */
    KnownErrorCode["AzureCognitiveSearchIndexLimitReached"] = "AzureCognitiveSearchIndexLimitReached";
    /** InternalServerError */
    KnownErrorCode["InternalServerError"] = "InternalServerError";
    /** ServiceUnavailable */
    KnownErrorCode["ServiceUnavailable"] = "ServiceUnavailable";
    /** Timeout */
    KnownErrorCode["Timeout"] = "Timeout";
    /** QuotaExceeded */
    KnownErrorCode["QuotaExceeded"] = "QuotaExceeded";
    /** Conflict */
    KnownErrorCode["Conflict"] = "Conflict";
    /** Warning */
    KnownErrorCode["Warning"] = "Warning";
})(KnownErrorCode || (KnownErrorCode = {}));
/** Known values of {@link InnerErrorCode} that the service accepts. */
export var KnownInnerErrorCode;
(function (KnownInnerErrorCode) {
    /** InvalidRequest */
    KnownInnerErrorCode["InvalidRequest"] = "InvalidRequest";
    /** InvalidParameterValue */
    KnownInnerErrorCode["InvalidParameterValue"] = "InvalidParameterValue";
    /** KnowledgeBaseNotFound */
    KnownInnerErrorCode["KnowledgeBaseNotFound"] = "KnowledgeBaseNotFound";
    /** AzureCognitiveSearchNotFound */
    KnownInnerErrorCode["AzureCognitiveSearchNotFound"] = "AzureCognitiveSearchNotFound";
    /** AzureCognitiveSearchThrottling */
    KnownInnerErrorCode["AzureCognitiveSearchThrottling"] = "AzureCognitiveSearchThrottling";
    /** ExtractionFailure */
    KnownInnerErrorCode["ExtractionFailure"] = "ExtractionFailure";
    /** InvalidRequestBodyFormat */
    KnownInnerErrorCode["InvalidRequestBodyFormat"] = "InvalidRequestBodyFormat";
    /** EmptyRequest */
    KnownInnerErrorCode["EmptyRequest"] = "EmptyRequest";
    /** MissingInputDocuments */
    KnownInnerErrorCode["MissingInputDocuments"] = "MissingInputDocuments";
    /** InvalidDocument */
    KnownInnerErrorCode["InvalidDocument"] = "InvalidDocument";
    /** ModelVersionIncorrect */
    KnownInnerErrorCode["ModelVersionIncorrect"] = "ModelVersionIncorrect";
    /** InvalidDocumentBatch */
    KnownInnerErrorCode["InvalidDocumentBatch"] = "InvalidDocumentBatch";
    /** UnsupportedLanguageCode */
    KnownInnerErrorCode["UnsupportedLanguageCode"] = "UnsupportedLanguageCode";
    /** InvalidCountryHint */
    KnownInnerErrorCode["InvalidCountryHint"] = "InvalidCountryHint";
})(KnownInnerErrorCode || (KnownInnerErrorCode = {}));
/** Known values of {@link AnalyzeTextLROTaskKind} that the service accepts. */
export var KnownAnalyzeTextLROTaskKind;
(function (KnownAnalyzeTextLROTaskKind) {
    /** SentimentAnalysis */
    KnownAnalyzeTextLROTaskKind["SentimentAnalysis"] = "SentimentAnalysis";
    /** EntityRecognition */
    KnownAnalyzeTextLROTaskKind["EntityRecognition"] = "EntityRecognition";
    /** PiiEntityRecognition */
    KnownAnalyzeTextLROTaskKind["PiiEntityRecognition"] = "PiiEntityRecognition";
    /** KeyPhraseExtraction */
    KnownAnalyzeTextLROTaskKind["KeyPhraseExtraction"] = "KeyPhraseExtraction";
    /** EntityLinking */
    KnownAnalyzeTextLROTaskKind["EntityLinking"] = "EntityLinking";
    /** Healthcare */
    KnownAnalyzeTextLROTaskKind["Healthcare"] = "Healthcare";
    /** ExtractiveSummarization */
    KnownAnalyzeTextLROTaskKind["ExtractiveSummarization"] = "ExtractiveSummarization";
    /** CustomEntityRecognition */
    KnownAnalyzeTextLROTaskKind["CustomEntityRecognition"] = "CustomEntityRecognition";
    /** CustomSingleLabelClassification */
    KnownAnalyzeTextLROTaskKind["CustomSingleLabelClassification"] = "CustomSingleLabelClassification";
    /** CustomMultiLabelClassification */
    KnownAnalyzeTextLROTaskKind["CustomMultiLabelClassification"] = "CustomMultiLabelClassification";
    /** AbstractiveSummarization */
    KnownAnalyzeTextLROTaskKind["AbstractiveSummarization"] = "AbstractiveSummarization";
})(KnownAnalyzeTextLROTaskKind || (KnownAnalyzeTextLROTaskKind = {}));
/** Known values of {@link OperationStatus} that the service accepts. */
export var KnownOperationStatus;
(function (KnownOperationStatus) {
    /** NotStarted */
    KnownOperationStatus["NotStarted"] = "notStarted";
    /** Running */
    KnownOperationStatus["Running"] = "running";
    /** Succeeded */
    KnownOperationStatus["Succeeded"] = "succeeded";
    /** PartiallyCompleted */
    KnownOperationStatus["PartiallyCompleted"] = "partiallyCompleted";
    /** Failed */
    KnownOperationStatus["Failed"] = "failed";
    /** Cancelled */
    KnownOperationStatus["Cancelled"] = "cancelled";
    /** Cancelling */
    KnownOperationStatus["Cancelling"] = "cancelling";
})(KnownOperationStatus || (KnownOperationStatus = {}));
/** Known values of {@link AnalyzeTextLROResultsKind} that the service accepts. */
export var KnownAnalyzeTextLROResultsKind;
(function (KnownAnalyzeTextLROResultsKind) {
    /** SentimentAnalysisLROResults */
    KnownAnalyzeTextLROResultsKind["SentimentAnalysisLROResults"] = "SentimentAnalysisLROResults";
    /** EntityRecognitionLROResults */
    KnownAnalyzeTextLROResultsKind["EntityRecognitionLROResults"] = "EntityRecognitionLROResults";
    /** PiiEntityRecognitionLROResults */
    KnownAnalyzeTextLROResultsKind["PiiEntityRecognitionLROResults"] = "PiiEntityRecognitionLROResults";
    /** KeyPhraseExtractionLROResults */
    KnownAnalyzeTextLROResultsKind["KeyPhraseExtractionLROResults"] = "KeyPhraseExtractionLROResults";
    /** EntityLinkingLROResults */
    KnownAnalyzeTextLROResultsKind["EntityLinkingLROResults"] = "EntityLinkingLROResults";
    /** HealthcareLROResults */
    KnownAnalyzeTextLROResultsKind["HealthcareLROResults"] = "HealthcareLROResults";
    /** ExtractiveSummarizationLROResults */
    KnownAnalyzeTextLROResultsKind["ExtractiveSummarizationLROResults"] = "ExtractiveSummarizationLROResults";
    /** CustomEntityRecognitionLROResults */
    KnownAnalyzeTextLROResultsKind["CustomEntityRecognitionLROResults"] = "CustomEntityRecognitionLROResults";
    /** CustomSingleLabelClassificationLROResults */
    KnownAnalyzeTextLROResultsKind["CustomSingleLabelClassificationLROResults"] = "CustomSingleLabelClassificationLROResults";
    /** CustomMultiLabelClassificationLROResults */
    KnownAnalyzeTextLROResultsKind["CustomMultiLabelClassificationLROResults"] = "CustomMultiLabelClassificationLROResults";
    /** AbstractiveSummarizationLROResults */
    KnownAnalyzeTextLROResultsKind["AbstractiveSummarizationLROResults"] = "AbstractiveSummarizationLROResults";
})(KnownAnalyzeTextLROResultsKind || (KnownAnalyzeTextLROResultsKind = {}));
/** Known values of {@link State} that the service accepts. */
export var KnownState;
(function (KnownState) {
    /** NotStarted */
    KnownState["NotStarted"] = "notStarted";
    /** Running */
    KnownState["Running"] = "running";
    /** Succeeded */
    KnownState["Succeeded"] = "succeeded";
    /** Failed */
    KnownState["Failed"] = "failed";
    /** Cancelled */
    KnownState["Cancelled"] = "cancelled";
    /** Cancelling */
    KnownState["Cancelling"] = "cancelling";
})(KnownState || (KnownState = {}));
/** Known values of {@link StringIndexType} that the service accepts. */
export var KnownStringIndexType;
(function (KnownStringIndexType) {
    /** Returned offset and length values will correspond to TextElements (Graphemes and Grapheme clusters) confirming to the Unicode 8.0.0 standard. Use this option if your application is written in .Net Framework or .Net Core and you will be using StringInfo. */
    KnownStringIndexType["TextElementsV8"] = "TextElements_v8";
    /** Returned offset and length values will correspond to Unicode code points. Use this option if your application is written in a language that support Unicode, for example Python. */
    KnownStringIndexType["UnicodeCodePoint"] = "UnicodeCodePoint";
    /** Returned offset and length values will correspond to UTF-16 code units. Use this option if your application is written in a language that support Unicode, for example Java, JavaScript. */
    KnownStringIndexType["Utf16CodeUnit"] = "Utf16CodeUnit";
})(KnownStringIndexType || (KnownStringIndexType = {}));
/** Known values of {@link PiiEntityDomain} that the service accepts. */
export var KnownPiiEntityDomain;
(function (KnownPiiEntityDomain) {
    /** Indicates that entities in the Personal Health Information domain should be redacted. */
    KnownPiiEntityDomain["Phi"] = "phi";
    /** Indicates that no domain is specified. */
    KnownPiiEntityDomain["None"] = "none";
})(KnownPiiEntityDomain || (KnownPiiEntityDomain = {}));
/** Known values of {@link PiiEntityCategory} that the service accepts. */
export var KnownPiiEntityCategory;
(function (KnownPiiEntityCategory) {
    /** ABARoutingNumber */
    KnownPiiEntityCategory["ABARoutingNumber"] = "ABARoutingNumber";
    /** ARNationalIdentityNumber */
    KnownPiiEntityCategory["ARNationalIdentityNumber"] = "ARNationalIdentityNumber";
    /** AUBankAccountNumber */
    KnownPiiEntityCategory["AUBankAccountNumber"] = "AUBankAccountNumber";
    /** AUDriversLicenseNumber */
    KnownPiiEntityCategory["AUDriversLicenseNumber"] = "AUDriversLicenseNumber";
    /** AUMedicalAccountNumber */
    KnownPiiEntityCategory["AUMedicalAccountNumber"] = "AUMedicalAccountNumber";
    /** AUPassportNumber */
    KnownPiiEntityCategory["AUPassportNumber"] = "AUPassportNumber";
    /** AUTaxFileNumber */
    KnownPiiEntityCategory["AUTaxFileNumber"] = "AUTaxFileNumber";
    /** AUBusinessNumber */
    KnownPiiEntityCategory["AUBusinessNumber"] = "AUBusinessNumber";
    /** AUCompanyNumber */
    KnownPiiEntityCategory["AUCompanyNumber"] = "AUCompanyNumber";
    /** ATIdentityCard */
    KnownPiiEntityCategory["ATIdentityCard"] = "ATIdentityCard";
    /** ATTaxIdentificationNumber */
    KnownPiiEntityCategory["ATTaxIdentificationNumber"] = "ATTaxIdentificationNumber";
    /** ATValueAddedTaxNumber */
    KnownPiiEntityCategory["ATValueAddedTaxNumber"] = "ATValueAddedTaxNumber";
    /** AzureDocumentDBAuthKey */
    KnownPiiEntityCategory["AzureDocumentDBAuthKey"] = "AzureDocumentDBAuthKey";
    /** AzureIaasDatabaseConnectionAndSQLString */
    KnownPiiEntityCategory["AzureIaasDatabaseConnectionAndSQLString"] = "AzureIAASDatabaseConnectionAndSQLString";
    /** AzureIoTConnectionString */
    KnownPiiEntityCategory["AzureIoTConnectionString"] = "AzureIoTConnectionString";
    /** AzurePublishSettingPassword */
    KnownPiiEntityCategory["AzurePublishSettingPassword"] = "AzurePublishSettingPassword";
    /** AzureRedisCacheString */
    KnownPiiEntityCategory["AzureRedisCacheString"] = "AzureRedisCacheString";
    /** AzureSAS */
    KnownPiiEntityCategory["AzureSAS"] = "AzureSAS";
    /** AzureServiceBusString */
    KnownPiiEntityCategory["AzureServiceBusString"] = "AzureServiceBusString";
    /** AzureStorageAccountKey */
    KnownPiiEntityCategory["AzureStorageAccountKey"] = "AzureStorageAccountKey";
    /** AzureStorageAccountGeneric */
    KnownPiiEntityCategory["AzureStorageAccountGeneric"] = "AzureStorageAccountGeneric";
    /** BENationalNumber */
    KnownPiiEntityCategory["BENationalNumber"] = "BENationalNumber";
    /** BENationalNumberV2 */
    KnownPiiEntityCategory["BENationalNumberV2"] = "BENationalNumberV2";
    /** BEValueAddedTaxNumber */
    KnownPiiEntityCategory["BEValueAddedTaxNumber"] = "BEValueAddedTaxNumber";
    /** BrcpfNumber */
    KnownPiiEntityCategory["BrcpfNumber"] = "BRCPFNumber";
    /** BRLegalEntityNumber */
    KnownPiiEntityCategory["BRLegalEntityNumber"] = "BRLegalEntityNumber";
    /** BRNationalIdrg */
    KnownPiiEntityCategory["BRNationalIdrg"] = "BRNationalIDRG";
    /** BGUniformCivilNumber */
    KnownPiiEntityCategory["BGUniformCivilNumber"] = "BGUniformCivilNumber";
    /** CABankAccountNumber */
    KnownPiiEntityCategory["CABankAccountNumber"] = "CABankAccountNumber";
    /** CADriversLicenseNumber */
    KnownPiiEntityCategory["CADriversLicenseNumber"] = "CADriversLicenseNumber";
    /** CAHealthServiceNumber */
    KnownPiiEntityCategory["CAHealthServiceNumber"] = "CAHealthServiceNumber";
    /** CAPassportNumber */
    KnownPiiEntityCategory["CAPassportNumber"] = "CAPassportNumber";
    /** CAPersonalHealthIdentification */
    KnownPiiEntityCategory["CAPersonalHealthIdentification"] = "CAPersonalHealthIdentification";
    /** CASocialInsuranceNumber */
    KnownPiiEntityCategory["CASocialInsuranceNumber"] = "CASocialInsuranceNumber";
    /** CLIdentityCardNumber */
    KnownPiiEntityCategory["CLIdentityCardNumber"] = "CLIdentityCardNumber";
    /** CNResidentIdentityCardNumber */
    KnownPiiEntityCategory["CNResidentIdentityCardNumber"] = "CNResidentIdentityCardNumber";
    /** CreditCardNumber */
    KnownPiiEntityCategory["CreditCardNumber"] = "CreditCardNumber";
    /** HRIdentityCardNumber */
    KnownPiiEntityCategory["HRIdentityCardNumber"] = "HRIdentityCardNumber";
    /** HRNationalIDNumber */
    KnownPiiEntityCategory["HRNationalIDNumber"] = "HRNationalIDNumber";
    /** HRPersonalIdentificationNumber */
    KnownPiiEntityCategory["HRPersonalIdentificationNumber"] = "HRPersonalIdentificationNumber";
    /** HRPersonalIdentificationOIBNumberV2 */
    KnownPiiEntityCategory["HRPersonalIdentificationOIBNumberV2"] = "HRPersonalIdentificationOIBNumberV2";
    /** CYIdentityCard */
    KnownPiiEntityCategory["CYIdentityCard"] = "CYIdentityCard";
    /** CYTaxIdentificationNumber */
    KnownPiiEntityCategory["CYTaxIdentificationNumber"] = "CYTaxIdentificationNumber";
    /** CZPersonalIdentityNumber */
    KnownPiiEntityCategory["CZPersonalIdentityNumber"] = "CZPersonalIdentityNumber";
    /** CZPersonalIdentityV2 */
    KnownPiiEntityCategory["CZPersonalIdentityV2"] = "CZPersonalIdentityV2";
    /** DKPersonalIdentificationNumber */
    KnownPiiEntityCategory["DKPersonalIdentificationNumber"] = "DKPersonalIdentificationNumber";
    /** DKPersonalIdentificationV2 */
    KnownPiiEntityCategory["DKPersonalIdentificationV2"] = "DKPersonalIdentificationV2";
    /** DrugEnforcementAgencyNumber */
    KnownPiiEntityCategory["DrugEnforcementAgencyNumber"] = "DrugEnforcementAgencyNumber";
    /** EEPersonalIdentificationCode */
    KnownPiiEntityCategory["EEPersonalIdentificationCode"] = "EEPersonalIdentificationCode";
    /** EUDebitCardNumber */
    KnownPiiEntityCategory["EUDebitCardNumber"] = "EUDebitCardNumber";
    /** EUDriversLicenseNumber */
    KnownPiiEntityCategory["EUDriversLicenseNumber"] = "EUDriversLicenseNumber";
    /** EugpsCoordinates */
    KnownPiiEntityCategory["EugpsCoordinates"] = "EUGPSCoordinates";
    /** EUNationalIdentificationNumber */
    KnownPiiEntityCategory["EUNationalIdentificationNumber"] = "EUNationalIdentificationNumber";
    /** EUPassportNumber */
    KnownPiiEntityCategory["EUPassportNumber"] = "EUPassportNumber";
    /** EUSocialSecurityNumber */
    KnownPiiEntityCategory["EUSocialSecurityNumber"] = "EUSocialSecurityNumber";
    /** EUTaxIdentificationNumber */
    KnownPiiEntityCategory["EUTaxIdentificationNumber"] = "EUTaxIdentificationNumber";
    /** FIEuropeanHealthNumber */
    KnownPiiEntityCategory["FIEuropeanHealthNumber"] = "FIEuropeanHealthNumber";
    /** FINationalID */
    KnownPiiEntityCategory["FINationalID"] = "FINationalID";
    /** FINationalIDV2 */
    KnownPiiEntityCategory["FINationalIDV2"] = "FINationalIDV2";
    /** FIPassportNumber */
    KnownPiiEntityCategory["FIPassportNumber"] = "FIPassportNumber";
    /** FRDriversLicenseNumber */
    KnownPiiEntityCategory["FRDriversLicenseNumber"] = "FRDriversLicenseNumber";
    /** FRHealthInsuranceNumber */
    KnownPiiEntityCategory["FRHealthInsuranceNumber"] = "FRHealthInsuranceNumber";
    /** FRNationalID */
    KnownPiiEntityCategory["FRNationalID"] = "FRNationalID";
    /** FRPassportNumber */
    KnownPiiEntityCategory["FRPassportNumber"] = "FRPassportNumber";
    /** FRSocialSecurityNumber */
    KnownPiiEntityCategory["FRSocialSecurityNumber"] = "FRSocialSecurityNumber";
    /** FRTaxIdentificationNumber */
    KnownPiiEntityCategory["FRTaxIdentificationNumber"] = "FRTaxIdentificationNumber";
    /** FRValueAddedTaxNumber */
    KnownPiiEntityCategory["FRValueAddedTaxNumber"] = "FRValueAddedTaxNumber";
    /** DEDriversLicenseNumber */
    KnownPiiEntityCategory["DEDriversLicenseNumber"] = "DEDriversLicenseNumber";
    /** DEPassportNumber */
    KnownPiiEntityCategory["DEPassportNumber"] = "DEPassportNumber";
    /** DEIdentityCardNumber */
    KnownPiiEntityCategory["DEIdentityCardNumber"] = "DEIdentityCardNumber";
    /** DETaxIdentificationNumber */
    KnownPiiEntityCategory["DETaxIdentificationNumber"] = "DETaxIdentificationNumber";
    /** DEValueAddedNumber */
    KnownPiiEntityCategory["DEValueAddedNumber"] = "DEValueAddedNumber";
    /** GRNationalIDCard */
    KnownPiiEntityCategory["GRNationalIDCard"] = "GRNationalIDCard";
    /** GRNationalIDV2 */
    KnownPiiEntityCategory["GRNationalIDV2"] = "GRNationalIDV2";
    /** GRTaxIdentificationNumber */
    KnownPiiEntityCategory["GRTaxIdentificationNumber"] = "GRTaxIdentificationNumber";
    /** HKIdentityCardNumber */
    KnownPiiEntityCategory["HKIdentityCardNumber"] = "HKIdentityCardNumber";
    /** HUValueAddedNumber */
    KnownPiiEntityCategory["HUValueAddedNumber"] = "HUValueAddedNumber";
    /** HUPersonalIdentificationNumber */
    KnownPiiEntityCategory["HUPersonalIdentificationNumber"] = "HUPersonalIdentificationNumber";
    /** HUTaxIdentificationNumber */
    KnownPiiEntityCategory["HUTaxIdentificationNumber"] = "HUTaxIdentificationNumber";
    /** INPermanentAccount */
    KnownPiiEntityCategory["INPermanentAccount"] = "INPermanentAccount";
    /** INUniqueIdentificationNumber */
    KnownPiiEntityCategory["INUniqueIdentificationNumber"] = "INUniqueIdentificationNumber";
    /** IDIdentityCardNumber */
    KnownPiiEntityCategory["IDIdentityCardNumber"] = "IDIdentityCardNumber";
    /** InternationalBankingAccountNumber */
    KnownPiiEntityCategory["InternationalBankingAccountNumber"] = "InternationalBankingAccountNumber";
    /** IEPersonalPublicServiceNumber */
    KnownPiiEntityCategory["IEPersonalPublicServiceNumber"] = "IEPersonalPublicServiceNumber";
    /** IEPersonalPublicServiceNumberV2 */
    KnownPiiEntityCategory["IEPersonalPublicServiceNumberV2"] = "IEPersonalPublicServiceNumberV2";
    /** ILBankAccountNumber */
    KnownPiiEntityCategory["ILBankAccountNumber"] = "ILBankAccountNumber";
    /** ILNationalID */
    KnownPiiEntityCategory["ILNationalID"] = "ILNationalID";
    /** ITDriversLicenseNumber */
    KnownPiiEntityCategory["ITDriversLicenseNumber"] = "ITDriversLicenseNumber";
    /** ITFiscalCode */
    KnownPiiEntityCategory["ITFiscalCode"] = "ITFiscalCode";
    /** ITValueAddedTaxNumber */
    KnownPiiEntityCategory["ITValueAddedTaxNumber"] = "ITValueAddedTaxNumber";
    /** JPBankAccountNumber */
    KnownPiiEntityCategory["JPBankAccountNumber"] = "JPBankAccountNumber";
    /** JPDriversLicenseNumber */
    KnownPiiEntityCategory["JPDriversLicenseNumber"] = "JPDriversLicenseNumber";
    /** JPPassportNumber */
    KnownPiiEntityCategory["JPPassportNumber"] = "JPPassportNumber";
    /** JPResidentRegistrationNumber */
    KnownPiiEntityCategory["JPResidentRegistrationNumber"] = "JPResidentRegistrationNumber";
    /** JPSocialInsuranceNumber */
    KnownPiiEntityCategory["JPSocialInsuranceNumber"] = "JPSocialInsuranceNumber";
    /** JPMyNumberCorporate */
    KnownPiiEntityCategory["JPMyNumberCorporate"] = "JPMyNumberCorporate";
    /** JPMyNumberPersonal */
    KnownPiiEntityCategory["JPMyNumberPersonal"] = "JPMyNumberPersonal";
    /** JPResidenceCardNumber */
    KnownPiiEntityCategory["JPResidenceCardNumber"] = "JPResidenceCardNumber";
    /** LVPersonalCode */
    KnownPiiEntityCategory["LVPersonalCode"] = "LVPersonalCode";
    /** LTPersonalCode */
    KnownPiiEntityCategory["LTPersonalCode"] = "LTPersonalCode";
    /** LUNationalIdentificationNumberNatural */
    KnownPiiEntityCategory["LUNationalIdentificationNumberNatural"] = "LUNationalIdentificationNumberNatural";
    /** LUNationalIdentificationNumberNonNatural */
    KnownPiiEntityCategory["LUNationalIdentificationNumberNonNatural"] = "LUNationalIdentificationNumberNonNatural";
    /** MYIdentityCardNumber */
    KnownPiiEntityCategory["MYIdentityCardNumber"] = "MYIdentityCardNumber";
    /** MTIdentityCardNumber */
    KnownPiiEntityCategory["MTIdentityCardNumber"] = "MTIdentityCardNumber";
    /** MTTaxIDNumber */
    KnownPiiEntityCategory["MTTaxIDNumber"] = "MTTaxIDNumber";
    /** NLCitizensServiceNumber */
    KnownPiiEntityCategory["NLCitizensServiceNumber"] = "NLCitizensServiceNumber";
    /** NLCitizensServiceNumberV2 */
    KnownPiiEntityCategory["NLCitizensServiceNumberV2"] = "NLCitizensServiceNumberV2";
    /** NLTaxIdentificationNumber */
    KnownPiiEntityCategory["NLTaxIdentificationNumber"] = "NLTaxIdentificationNumber";
    /** NLValueAddedTaxNumber */
    KnownPiiEntityCategory["NLValueAddedTaxNumber"] = "NLValueAddedTaxNumber";
    /** NZBankAccountNumber */
    KnownPiiEntityCategory["NZBankAccountNumber"] = "NZBankAccountNumber";
    /** NZDriversLicenseNumber */
    KnownPiiEntityCategory["NZDriversLicenseNumber"] = "NZDriversLicenseNumber";
    /** NZInlandRevenueNumber */
    KnownPiiEntityCategory["NZInlandRevenueNumber"] = "NZInlandRevenueNumber";
    /** NZMinistryOfHealthNumber */
    KnownPiiEntityCategory["NZMinistryOfHealthNumber"] = "NZMinistryOfHealthNumber";
    /** NZSocialWelfareNumber */
    KnownPiiEntityCategory["NZSocialWelfareNumber"] = "NZSocialWelfareNumber";
    /** NOIdentityNumber */
    KnownPiiEntityCategory["NOIdentityNumber"] = "NOIdentityNumber";
    /** PHUnifiedMultiPurposeIDNumber */
    KnownPiiEntityCategory["PHUnifiedMultiPurposeIDNumber"] = "PHUnifiedMultiPurposeIDNumber";
    /** PLIdentityCard */
    KnownPiiEntityCategory["PLIdentityCard"] = "PLIdentityCard";
    /** PLNationalID */
    KnownPiiEntityCategory["PLNationalID"] = "PLNationalID";
    /** PLNationalIDV2 */
    KnownPiiEntityCategory["PLNationalIDV2"] = "PLNationalIDV2";
    /** PLPassportNumber */
    KnownPiiEntityCategory["PLPassportNumber"] = "PLPassportNumber";
    /** PLTaxIdentificationNumber */
    KnownPiiEntityCategory["PLTaxIdentificationNumber"] = "PLTaxIdentificationNumber";
    /** PlregonNumber */
    KnownPiiEntityCategory["PlregonNumber"] = "PLREGONNumber";
    /** PTCitizenCardNumber */
    KnownPiiEntityCategory["PTCitizenCardNumber"] = "PTCitizenCardNumber";
    /** PTCitizenCardNumberV2 */
    KnownPiiEntityCategory["PTCitizenCardNumberV2"] = "PTCitizenCardNumberV2";
    /** PTTaxIdentificationNumber */
    KnownPiiEntityCategory["PTTaxIdentificationNumber"] = "PTTaxIdentificationNumber";
    /** ROPersonalNumericalCode */
    KnownPiiEntityCategory["ROPersonalNumericalCode"] = "ROPersonalNumericalCode";
    /** RUPassportNumberDomestic */
    KnownPiiEntityCategory["RUPassportNumberDomestic"] = "RUPassportNumberDomestic";
    /** RUPassportNumberInternational */
    KnownPiiEntityCategory["RUPassportNumberInternational"] = "RUPassportNumberInternational";
    /** SANationalID */
    KnownPiiEntityCategory["SANationalID"] = "SANationalID";
    /** SGNationalRegistrationIdentityCardNumber */
    KnownPiiEntityCategory["SGNationalRegistrationIdentityCardNumber"] = "SGNationalRegistrationIdentityCardNumber";
    /** SKPersonalNumber */
    KnownPiiEntityCategory["SKPersonalNumber"] = "SKPersonalNumber";
    /** SITaxIdentificationNumber */
    KnownPiiEntityCategory["SITaxIdentificationNumber"] = "SITaxIdentificationNumber";
    /** SIUniqueMasterCitizenNumber */
    KnownPiiEntityCategory["SIUniqueMasterCitizenNumber"] = "SIUniqueMasterCitizenNumber";
    /** ZAIdentificationNumber */
    KnownPiiEntityCategory["ZAIdentificationNumber"] = "ZAIdentificationNumber";
    /** KRResidentRegistrationNumber */
    KnownPiiEntityCategory["KRResidentRegistrationNumber"] = "KRResidentRegistrationNumber";
    /** Esdni */
    KnownPiiEntityCategory["Esdni"] = "ESDNI";
    /** ESSocialSecurityNumber */
    KnownPiiEntityCategory["ESSocialSecurityNumber"] = "ESSocialSecurityNumber";
    /** ESTaxIdentificationNumber */
    KnownPiiEntityCategory["ESTaxIdentificationNumber"] = "ESTaxIdentificationNumber";
    /** SQLServerConnectionString */
    KnownPiiEntityCategory["SQLServerConnectionString"] = "SQLServerConnectionString";
    /** SENationalID */
    KnownPiiEntityCategory["SENationalID"] = "SENationalID";
    /** SENationalIDV2 */
    KnownPiiEntityCategory["SENationalIDV2"] = "SENationalIDV2";
    /** SEPassportNumber */
    KnownPiiEntityCategory["SEPassportNumber"] = "SEPassportNumber";
    /** SETaxIdentificationNumber */
    KnownPiiEntityCategory["SETaxIdentificationNumber"] = "SETaxIdentificationNumber";
    /** SwiftCode */
    KnownPiiEntityCategory["SwiftCode"] = "SWIFTCode";
    /** CHSocialSecurityNumber */
    KnownPiiEntityCategory["CHSocialSecurityNumber"] = "CHSocialSecurityNumber";
    /** TWNationalID */
    KnownPiiEntityCategory["TWNationalID"] = "TWNationalID";
    /** TWPassportNumber */
    KnownPiiEntityCategory["TWPassportNumber"] = "TWPassportNumber";
    /** TWResidentCertificate */
    KnownPiiEntityCategory["TWResidentCertificate"] = "TWResidentCertificate";
    /** THPopulationIdentificationCode */
    KnownPiiEntityCategory["THPopulationIdentificationCode"] = "THPopulationIdentificationCode";
    /** TRNationalIdentificationNumber */
    KnownPiiEntityCategory["TRNationalIdentificationNumber"] = "TRNationalIdentificationNumber";
    /** UKDriversLicenseNumber */
    KnownPiiEntityCategory["UKDriversLicenseNumber"] = "UKDriversLicenseNumber";
    /** UKElectoralRollNumber */
    KnownPiiEntityCategory["UKElectoralRollNumber"] = "UKElectoralRollNumber";
    /** UKNationalHealthNumber */
    KnownPiiEntityCategory["UKNationalHealthNumber"] = "UKNationalHealthNumber";
    /** UKNationalInsuranceNumber */
    KnownPiiEntityCategory["UKNationalInsuranceNumber"] = "UKNationalInsuranceNumber";
    /** UKUniqueTaxpayerNumber */
    KnownPiiEntityCategory["UKUniqueTaxpayerNumber"] = "UKUniqueTaxpayerNumber";
    /** UsukPassportNumber */
    KnownPiiEntityCategory["UsukPassportNumber"] = "USUKPassportNumber";
    /** USBankAccountNumber */
    KnownPiiEntityCategory["USBankAccountNumber"] = "USBankAccountNumber";
    /** USDriversLicenseNumber */
    KnownPiiEntityCategory["USDriversLicenseNumber"] = "USDriversLicenseNumber";
    /** USIndividualTaxpayerIdentification */
    KnownPiiEntityCategory["USIndividualTaxpayerIdentification"] = "USIndividualTaxpayerIdentification";
    /** USSocialSecurityNumber */
    KnownPiiEntityCategory["USSocialSecurityNumber"] = "USSocialSecurityNumber";
    /** UAPassportNumberDomestic */
    KnownPiiEntityCategory["UAPassportNumberDomestic"] = "UAPassportNumberDomestic";
    /** UAPassportNumberInternational */
    KnownPiiEntityCategory["UAPassportNumberInternational"] = "UAPassportNumberInternational";
    /** Organization */
    KnownPiiEntityCategory["Organization"] = "Organization";
    /** Email */
    KnownPiiEntityCategory["Email"] = "Email";
    /** URL */
    KnownPiiEntityCategory["URL"] = "URL";
    /** Age */
    KnownPiiEntityCategory["Age"] = "Age";
    /** PhoneNumber */
    KnownPiiEntityCategory["PhoneNumber"] = "PhoneNumber";
    /** IPAddress */
    KnownPiiEntityCategory["IPAddress"] = "IPAddress";
    /** Date */
    KnownPiiEntityCategory["Date"] = "Date";
    /** Person */
    KnownPiiEntityCategory["Person"] = "Person";
    /** Address */
    KnownPiiEntityCategory["Address"] = "Address";
    /** All */
    KnownPiiEntityCategory["All"] = "All";
    /** Default */
    KnownPiiEntityCategory["Default"] = "Default";
})(KnownPiiEntityCategory || (KnownPiiEntityCategory = {}));
/** Known values of {@link WarningCode} that the service accepts. */
export var KnownWarningCode;
(function (KnownWarningCode) {
    /** LongWordsInDocument */
    KnownWarningCode["LongWordsInDocument"] = "LongWordsInDocument";
    /** DocumentTruncated */
    KnownWarningCode["DocumentTruncated"] = "DocumentTruncated";
})(KnownWarningCode || (KnownWarningCode = {}));
/** Known values of {@link HealthcareEntityCategory} that the service accepts. */
export var KnownHealthcareEntityCategory;
(function (KnownHealthcareEntityCategory) {
    /** BodyStructure */
    KnownHealthcareEntityCategory["BodyStructure"] = "BodyStructure";
    /** Age */
    KnownHealthcareEntityCategory["Age"] = "Age";
    /** Gender */
    KnownHealthcareEntityCategory["Gender"] = "Gender";
    /** Ethnicity */
    KnownHealthcareEntityCategory["Ethnicity"] = "Ethnicity";
    /** ExaminationName */
    KnownHealthcareEntityCategory["ExaminationName"] = "ExaminationName";
    /** Date */
    KnownHealthcareEntityCategory["Date"] = "Date";
    /** Direction */
    KnownHealthcareEntityCategory["Direction"] = "Direction";
    /** Frequency */
    KnownHealthcareEntityCategory["Frequency"] = "Frequency";
    /** MeasurementValue */
    KnownHealthcareEntityCategory["MeasurementValue"] = "MeasurementValue";
    /** MeasurementUnit */
    KnownHealthcareEntityCategory["MeasurementUnit"] = "MeasurementUnit";
    /** RelationalOperator */
    KnownHealthcareEntityCategory["RelationalOperator"] = "RelationalOperator";
    /** Time */
    KnownHealthcareEntityCategory["Time"] = "Time";
    /** Course */
    KnownHealthcareEntityCategory["Course"] = "Course";
    /** GeneOrProtein */
    KnownHealthcareEntityCategory["GeneOrProtein"] = "GeneOrProtein";
    /** Variant */
    KnownHealthcareEntityCategory["Variant"] = "Variant";
    /** Expression */
    KnownHealthcareEntityCategory["Expression"] = "Expression";
    /** MutationType */
    KnownHealthcareEntityCategory["MutationType"] = "MutationType";
    /** AdministrativeEvent */
    KnownHealthcareEntityCategory["AdministrativeEvent"] = "AdministrativeEvent";
    /** CareEnvironment */
    KnownHealthcareEntityCategory["CareEnvironment"] = "CareEnvironment";
    /** HealthcareProfession */
    KnownHealthcareEntityCategory["HealthcareProfession"] = "HealthcareProfession";
    /** Diagnosis */
    KnownHealthcareEntityCategory["Diagnosis"] = "Diagnosis";
    /** SymptomOrSign */
    KnownHealthcareEntityCategory["SymptomOrSign"] = "SymptomOrSign";
    /** ConditionQualifier */
    KnownHealthcareEntityCategory["ConditionQualifier"] = "ConditionQualifier";
    /** ConditionScale */
    KnownHealthcareEntityCategory["ConditionScale"] = "ConditionScale";
    /** MedicationClass */
    KnownHealthcareEntityCategory["MedicationClass"] = "MedicationClass";
    /** MedicationName */
    KnownHealthcareEntityCategory["MedicationName"] = "MedicationName";
    /** Dosage */
    KnownHealthcareEntityCategory["Dosage"] = "Dosage";
    /** MedicationForm */
    KnownHealthcareEntityCategory["MedicationForm"] = "MedicationForm";
    /** MedicationRoute */
    KnownHealthcareEntityCategory["MedicationRoute"] = "MedicationRoute";
    /** FamilyRelation */
    KnownHealthcareEntityCategory["FamilyRelation"] = "FamilyRelation";
    /** TreatmentName */
    KnownHealthcareEntityCategory["TreatmentName"] = "TreatmentName";
    /** Allergen */
    KnownHealthcareEntityCategory["Allergen"] = "Allergen";
    /** Employment */
    KnownHealthcareEntityCategory["Employment"] = "Employment";
    /** LivingStatus */
    KnownHealthcareEntityCategory["LivingStatus"] = "LivingStatus";
    /** SubstanceUse */
    KnownHealthcareEntityCategory["SubstanceUse"] = "SubstanceUse";
    /** SubstanceUseAmount */
    KnownHealthcareEntityCategory["SubstanceUseAmount"] = "SubstanceUseAmount";
})(KnownHealthcareEntityCategory || (KnownHealthcareEntityCategory = {}));
/** Known values of {@link RelationType} that the service accepts. */
export var KnownRelationType;
(function (KnownRelationType) {
    /** Abbreviation */
    KnownRelationType["Abbreviation"] = "Abbreviation";
    /** BodySiteOfCondition */
    KnownRelationType["BodySiteOfCondition"] = "BodySiteOfCondition";
    /** BodySiteOfTreatment */
    KnownRelationType["BodySiteOfTreatment"] = "BodySiteOfTreatment";
    /** CourseOfCondition */
    KnownRelationType["CourseOfCondition"] = "CourseOfCondition";
    /** CourseOfExamination */
    KnownRelationType["CourseOfExamination"] = "CourseOfExamination";
    /** CourseOfMedication */
    KnownRelationType["CourseOfMedication"] = "CourseOfMedication";
    /** CourseOfTreatment */
    KnownRelationType["CourseOfTreatment"] = "CourseOfTreatment";
    /** DirectionOfBodyStructure */
    KnownRelationType["DirectionOfBodyStructure"] = "DirectionOfBodyStructure";
    /** DirectionOfCondition */
    KnownRelationType["DirectionOfCondition"] = "DirectionOfCondition";
    /** DirectionOfExamination */
    KnownRelationType["DirectionOfExamination"] = "DirectionOfExamination";
    /** DirectionOfTreatment */
    KnownRelationType["DirectionOfTreatment"] = "DirectionOfTreatment";
    /** DosageOfMedication */
    KnownRelationType["DosageOfMedication"] = "DosageOfMedication";
    /** ExaminationFindsCondition */
    KnownRelationType["ExaminationFindsCondition"] = "ExaminationFindsCondition";
    /** ExpressionOfGene */
    KnownRelationType["ExpressionOfGene"] = "ExpressionOfGene";
    /** ExpressionOfVariant */
    KnownRelationType["ExpressionOfVariant"] = "ExpressionOfVariant";
    /** FormOfMedication */
    KnownRelationType["FormOfMedication"] = "FormOfMedication";
    /** FrequencyOfCondition */
    KnownRelationType["FrequencyOfCondition"] = "FrequencyOfCondition";
    /** FrequencyOfMedication */
    KnownRelationType["FrequencyOfMedication"] = "FrequencyOfMedication";
    /** FrequencyOfTreatment */
    KnownRelationType["FrequencyOfTreatment"] = "FrequencyOfTreatment";
    /** MutationTypeOfGene */
    KnownRelationType["MutationTypeOfGene"] = "MutationTypeOfGene";
    /** MutationTypeOfVariant */
    KnownRelationType["MutationTypeOfVariant"] = "MutationTypeOfVariant";
    /** QualifierOfCondition */
    KnownRelationType["QualifierOfCondition"] = "QualifierOfCondition";
    /** RelationOfExamination */
    KnownRelationType["RelationOfExamination"] = "RelationOfExamination";
    /** RouteOfMedication */
    KnownRelationType["RouteOfMedication"] = "RouteOfMedication";
    /** ScaleOfCondition */
    KnownRelationType["ScaleOfCondition"] = "ScaleOfCondition";
    /** TimeOfCondition */
    KnownRelationType["TimeOfCondition"] = "TimeOfCondition";
    /** TimeOfEvent */
    KnownRelationType["TimeOfEvent"] = "TimeOfEvent";
    /** TimeOfExamination */
    KnownRelationType["TimeOfExamination"] = "TimeOfExamination";
    /** TimeOfMedication */
    KnownRelationType["TimeOfMedication"] = "TimeOfMedication";
    /** TimeOfTreatment */
    KnownRelationType["TimeOfTreatment"] = "TimeOfTreatment";
    /** UnitOfCondition */
    KnownRelationType["UnitOfCondition"] = "UnitOfCondition";
    /** UnitOfExamination */
    KnownRelationType["UnitOfExamination"] = "UnitOfExamination";
    /** ValueOfCondition */
    KnownRelationType["ValueOfCondition"] = "ValueOfCondition";
    /** ValueOfExamination */
    KnownRelationType["ValueOfExamination"] = "ValueOfExamination";
    /** VariantOfGene */
    KnownRelationType["VariantOfGene"] = "VariantOfGene";
})(KnownRelationType || (KnownRelationType = {}));
/** Known values of {@link ExtractiveSummarizationOrderingCriteria} that the service accepts. */
export var KnownExtractiveSummarizationOrderingCriteria;
(function (KnownExtractiveSummarizationOrderingCriteria) {
    /** Indicates that results should be sorted in order of appearance in the text. */
    KnownExtractiveSummarizationOrderingCriteria["Offset"] = "Offset";
    /** Indicates that results should be sorted in order of importance (i.e. rank score) according to the model. */
    KnownExtractiveSummarizationOrderingCriteria["Rank"] = "Rank";
})(KnownExtractiveSummarizationOrderingCriteria || (KnownExtractiveSummarizationOrderingCriteria = {}));
